ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,671 @@
+    m_message = QString("Loading the utility network...");
+    emit messageChanged();
+
+    m_utilityNetwork = m_map->utilityNetworks()->first();
+    m_utilityNetwork->load();
+
+    m_serviceGeodatabase = m_utilityNetwork->serviceGeodatabase();
+    const QString uuid = QUuid().createUuid().toString(QUuid::WithoutBraces);
+
+    // Restrict editing and tracing on a random branch
+    ServiceVersionParameters* params = new ServiceVersionParameters(this);
+    params->setName("ValidateNetworkTopology_"+uuid);
+    params->setAccess(VersionAccess::Private);
+    params->setDescription("Validate network topology with ArcGIS Runtime");
+
+    connect(m_serviceGeodatabase, &ServiceGeodatabase::doneLoading, this, [this, params]()
+    {
+      if (m_serviceGeodatabase->loadStatus() == LoadStatus::Loaded)
+      {
+        m_serviceGeodatabase->createVersionAsync(params).then(this, [this](ServiceVersionInfo* serviceVersionInfo)
+        {
+          m_serviceGeodatabase->switchVersionAsync(serviceVersionInfo->name()).then(this, [this]()
+          {
+            getLabelDefinitions();
+
+            // Visualize dirty area by adding to the map
+            m_dirtyAreaTable = m_utilityNetwork->dirtyAreaTable();
+            m_map->operationalLayers()->append(new FeatureLayer(dynamic_cast<FeatureTable*>(m_dirtyAreaTable)));
+            });
+          }).onFailed(this, []()
+          {
+          });
+      }
+    });
+
+    connect(m_utilityNetwork, &UtilityNetwork::loadStatusChanged, this, [this]()
+    {
+      if (m_utilityNetwork->loadStatus() == LoadStatus::Loaded)
+      {
+        setupTraceParameters();
+      }
+    });
+  });
+}
+
+
+
+ValidateUtilityNetworkTopology::~ValidateUtilityNetworkTopology() = default;
+
+void ValidateUtilityNetworkTopology::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ValidateUtilityNetworkTopology>("Esri.Samples", 1, 0, "ValidateUtilityNetworkTopologySample");
+}
+
+MapQuickView* ValidateUtilityNetworkTopology::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ValidateUtilityNetworkTopology::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_busy = true;
+  emit isBusy();
+  m_message = QString("Loading a webmap...");
+  emit messageChanged();
+
+  Envelope envelope(
+      -9815489.0660101417,
+      5128463.4221229386,
+      -9814625.2768726498,
+      5128968.4911854975,
+      SpatialReference::webMercator()
+      );
+
+  Viewpoint initialViewpoint(envelope);
+
+  m_map->setInitialViewpoint(initialViewpoint);
+
+  // Load in persistent session mode (workaround for server caching issue)
+  // https://support.esri.com/en-us/bug/asynchronous-validate-request-for-utility-network-servi-bug-000160443
+  m_map->loadSettings()->setFeatureServiceSessionType(FeatureServiceSessionType(FeatureServiceSessionType::Persistent));
```

Comment:
Please confirm that the design mentions these comments should be present.

Also, it looks like you have unnecessary cast.

```suggestion
  // Load in persistent session mode (workaround for server caching issue)
  // https://support.esri.com/en-us/bug/asynchronous-validate-request-for-utility-network-servi-bug-000160443
  m_map->loadSettings()->setFeatureServiceSessionType(FeatureServiceSessionType::Persistent);
```

Reply:
comments are present in the [implementation PR](https://devtopia.esri.com/runtime/common-samples/pull/3889/files#diff-4fe070fdeccde9955a0571944b4a58ea29c7f3dbf49c1c89efc7392aaa6dce12R135)